src/pocketmine/entity/Human.php

```diff
@@ -117,10 +120,16 @@ public function __construct(Level $level, CompoundTag $nbt){
 	protected static function deserializeSkinNBT(CompoundTag $skinTag) : Skin{
 		$skin = new Skin(
 			$skinTag->getString("Name"),
-			$skinTag->hasTag("Data", StringTag::class) ? $skinTag->getString("Data") : $skinTag->getByteArray("Data"), //old data (this used to be saved as a StringTag in older versions of PM)
-			$skinTag->getByteArray("CapeData", ""),
-			$skinTag->getString("GeometryName", ""),
-			$skinTag->getByteArray("GeometryData", "")
+			$skinTag->getString("SkinResourcePatch", ""),
+			new SerializedImage($skinTag->getInt("SkinImageHeight"), $skinTag->getInt("SkinImageWidth"), $skinTag->getByteArray("Data")),
```

Comment:
`SkinImageHeight`, `SkinImageWidth` won't exist in worlds loaded from 1.12. This will cause a crash.

There is a backwards compatibility patch for `Data` that has been removed here that will also cause crashes on old data.